gr-azure-software-radio/python/qa_rest_api.py

```diff
@@ -0,0 +1,84 @@
+# pylint: disable=missing-function-docstring, no-self-use, missing-class-docstring, no-member, eval-used, unused-variable, attribute-defined-outside-init
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the GNU General Public License v3.0 or later.
+# See License.txt in the project root for license information.
+#
+
+import time
+from socket import socket
+import requests
+from gnuradio import gr_unittest
+from rest_api import RestApi
+
+class QaRestApi(gr_unittest.TestCase):
+
+    def get_free_port(self):
+        with socket() as sock:
+            sock.bind(('', 0))
+            return sock.getsockname()[1]
+
+    def test_instance(self):
+        port = self.get_free_port()
+        instance = RestApi(self, port)
+        self.assertIsNotNone(instance)
+
+    def test_get_vars(self):
+        self.check_var1 = 1
+        self.check_var2 = '2'
+        self.check_var3 = {'pi':3.14}
+        port = self.get_free_port()
+        instance = RestApi(self, port)
+        # give server a second to startup
+        time.sleep(1)
```

Comment:
What do you think about using the fastAPI TestClient for very focused tests on our endpoints? Might help with simplifying some of the low level tests, ensuring we return appropriate error codes, etc